roachtest: sqlsmith/setup=tpcc/setting=ddl-nodrop failed

roachtest.sqlsmith/setup=tpcc/setting=ddl-nodrop [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6079423?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6079423?buildTab=artifacts#/sqlsmith/setup=tpcc/setting=ddl-nodrop) on master @ [0dd438d3dc0b42543890455945a7a6b42811def1](https://github.com/cockroachdb/cockroach/commits/0dd438d3dc0b42543890455945a7a6b42811def1):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=tpcc/setting=ddl-nodrop/run_1
	sqlsmith.go:265,sqlsmith.go:325,test_runner.go:896: error: pq: internal error: failed to find column
		stmt:
		ALTER TABLE defaultdb.public.item DROP COLUMN tableoid CASCADE;
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=tpcc/setting=ddl-nodrop.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18556

roachtest.sqlsmith/setup=tpcc/setting=ddl-nodrop [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6107800?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6107800?buildTab=artifacts#/sqlsmith/setup=tpcc/setting=ddl-nodrop) on master @ [f4042d47fa8062a612c38d4696eb6bee9cee7c21](https://github.com/cockroachdb/cockroach/commits/f4042d47fa8062a612c38d4696eb6bee9cee7c21):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=tpcc/setting=ddl-nodrop/run_1
	sqlsmith.go:265,sqlsmith.go:325,test_runner.go:896: error: pq: internal error: failed to find column
		stmt:
		ALTER TABLE defaultdb.public.district DROP COLUMN tableoid RESTRICT;
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=tpcc/setting=ddl-nodrop.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/internal/sqlsmith/alter.go
+++ b/pkg/internal/sqlsmith/alter.go
@@ -254,7 +254,19 @@ func makeDropColumn(s *Smither) (tree.Statement, bool) {
 	if !ok {
 		return nil, false
 	}
-	col := tableRef.Columns[s.rnd.Intn(len(tableRef.Columns))]
+	// Pick a random column to drop while ignoring the system columns since
+	// those cannot be dropped.
+	var col *tree.ColumnTableDef
+	for {
+		col = tableRef.Columns[s.rnd.Intn(len(tableRef.Columns))]
+		var isSystemCol bool
+		for _, systemColDesc := range colinfo.AllSystemColumnDescs {
+			isSystemCol = isSystemCol || string(col.Name) == systemColDesc.Name
+		}
+		if !isSystemCol {
+			break
+		}
+	}
 
 	return &tree.AlterTable{
 		Table: tableRef.TableName.ToUnresolvedObjectName(),

